### DISCUSSION TITLE
What is format preservation, and how can I disable it?

### DISCUSSION THREAD

**Discussion by jennydaman** [Enet4/dicom-rs#401] [Q&A] [answered]
I see here that some element values are represented as `PrimitiveValue::Strs` even if their VR is something like "Decimal String".

https://github.com/Enet4/dicom-rs/blob/41e87f38ec977e1c92027b272278eb87e40d5718/core/src/value/primitive.rs#L226-L230

How can I read a DICOM file without format preservation?

**Comment by jennydaman** [comment]
I see that it's related to `ValueReadStrategy`

https://github.com/Enet4/dicom-rs/blob/41e87f38ec977e1c92027b272278eb87e40d5718/parser/src/dataset/read.rs#L89-L124

Is there any way I can pass a strategy to `dicom::object::open_file`?

**Comment by Enet4** [comment, marked as answer]
Yes, "format preservation" refers to the parser's value reading strategy that preserves textual values (numbers and dates) as strings. It is currently only available as an option in the `dicom_parser` data set reader API, as the use of any other strategy in the object API could bring other issues in the loading process. Namely, the strategy that reads everything as bytes has never really been tried here, and the strategy to convert textual numbers and dates would often choke on non-compliant or unsupported DICOM data.

It would be great if you could share the use case for changing the value reading strategy. It is generally sufficient to use one of the various `to_*` methods in the DICOM value types, which convert the underlying representation on demand into another, more usable type.
